cmd/uniq/uniq.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+)
+
+type options struct {
+	printDuplicates bool
+	printEmptyLines bool
+	printLineNumber bool
+}
+
+func parseArgs(args []string) options {
+	var opts options
+
+	for _, opt := range args[1:] {
+		switch opt {
+		case "-dup":
+			opts.printDuplicates = true
+		case "-empty":
+			opts.printEmptyLines = true
+		case "-num":
+			opts.printLineNumber = true
+		default:
+			fmt.Println("Usage:")
+			fmt.Println("uniq [-dup | -empty | -num]")
+			os.Exit(1)
+		}
+	}
+	return opts
+}
+
+func uniq(input *os.File, opts options) {
+	inputMap := make(map[string]*string)
+	lineNumsMap := make(map[*string][]int)
```

Comment:
Let us see if the `linesOrdered` slice suggested in https://github.com/c0defellas/enzo/pull/28#issuecomment-270253257 can suplant those maps.